File: src/Moryx.FactoryMonitor.Web/src/app/api/models/Moryx/FactoryMonitor/Endpoints/Models/resource-changed-model.ts
/* tslint:disable */
/* eslint-disable */
import { CellLocationModel as MoryxFactoryMonitorEndpointsModelCellLocationModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/cell-location-model';
import { CellPropertySettings as MoryxFactoryMonitorEndpointsModelCellPropertySettings } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/cell-property-settings';
export interface ResourceChangedModel {
  cellIconName?: null | string;
  cellImageURL?: null | string;
  cellLocation?: MoryxFactoryMonitorEndpointsModelCellLocationModel;
  cellName?: null | string;
  cellPropertySettings?: null | {
[key: string]: MoryxFactoryMonitorEndpointsModelCellPropertySettings;
};
  factoryId?: number;
  iconName?: null | string;
  id?: number;
  isACell?: boolean;
  location?: MoryxFactoryMonitorEndpointsModelCellLocationModel;
}

File: src/Moryx.FactoryMonitor.Endpoints/Models/ResourceChangedModel.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;

namespace Moryx.FactoryMonitor.Endpoints.Models;

/// <summary>
/// Model for the cell information for every ResourceUpdated event.
/// </summary>
[DataContract]
public class ResourceChangedModel : VisualizableItemModel
{
    [DataMember]
    public virtual string CellName { get; set; }

    // ToDo: Verify why this is duplicated and not using the location of the visual item
    [DataMember]
    public virtual string CellIconName { get => base.IconName; set => base.IconName = value; }

    [DataMember]
    public virtual string CellImageURL { get; set; }

    // ToDo: Verify why this is duplicated and not using the location of the visual item
    [DataMember]
    public long Id { get => base.Id; set => base.Id = value; }

    // ToDo: Verify why this is duplicated and not using the location of the visual item
    [DataMember]
    public CellLocationModel CellLocation { get => base.Location; set => base.Location = value; }

    [DataMember]
    public Dictionary<string, CellPropertySettings> CellPropertySettings { get; set; }

    [DataMember]
    public long FactoryId { get; internal set; }
}
